### PR TITLE
Predict inventory slot interactions.

### PR DESCRIPTION
**Pull request by ElectroJr**
This PR makes clicking on inventory slots predicted. This includes equipping, unequipping, and interact-using interactions, though this last bit is currently only useful for things like eject/insert interactions (e.g., pda).

Most of the changes involve just Combining `TryEquipNetworkMessage`, `TryUnequipNetworkMessage`, and `UseSlotNetworkMessage` into a single `UseSlotNetworkMessage`.

This PR also adds some interaction checks when equipping or unequipping items:
- Make sure items & target can be reached by the actor
- When handling the various `NetworkMessage`s, make sure the requested entity is actually the entity that sent the message (a bad actor could have remotely stripped everyone on the server)

### CODE REVIEW

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 1]
```diff
@@ -102,6 +159,28 @@ public virtual bool TryEquip(EntityUid actor, EntityUid target, EntityUid itemUi
         return true;
     }
 
+    public bool CanAccess(EntityUid actor, EntityUid target, EntityUid itemUid)
+    {
+        // Can the actor reach the target?
+        if (actor != target && !( actor.InRangeUnobstructed(target) && actor.IsInSameOrParentContainer(target)))
```
Comment (metalgearsloth) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:165]: Does containersystem have equivalents for IsInSameOrParentContainer?

**Review by metalgearsloth — APPROVED** [review]
Need to check in-game with lag

**Comment by metalgearsloth** [comment]
LGTM, would also be nice for 2nd review. Some edge cases with mispredicts but that's mainly on other systems to be predicted too (e.g. if you get slipped)

File: Content.Shared/Inventory/InventorySystem.Equip.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Equip.cs:165]: Not AFAIK. but I can open a PR to add it.

**Review by PaulRitter — CHANGES_REQUESTED** [review]
this has a pretty relevant regression in it, unless im missing something: the reason for my previous implementation of *equip* prediction was that client-side calls of try(un)equip get mirror to the server, if the entities arent clientside. afaict this removes that completely, meaning the only way to call tryequip clientside to have it replicated on server is through the useslot function, which is bad imo

**Comment by ElectroJr** [comment]
@PaulRitter 

AFAIK for prediction to work properly, the network-event raising has become a `RaisePredictiveEvent` call, and that has to be done outside of the client's `TryEquip` override function to avoid recursion.

So yes, this change does mean that any client-exclusive code / function-calls that result in the `TryEquip` calls won't automatically get sent to the server. This shouldn't be a problem for equipping due to key bindings  (e.g. `Z` for quick-equipping held clothing), as those key bindings should already be automatically sent to the server.

Other than inventory slots, the only instance I can think of where the client MIGHT need to initiate a `TryEquip` would be for whenever stripping/BUIs are predicted, but there I assume the bui-messages themselves would be raised as predictive events. If there are likely to be other times it would be useful, I can just re-add the `TryEquipNetworkMessage`s & subscriptions so people could directly raise those as predictive events.

File: Content.Shared/Inventory/InventorySystem.Equip.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Inventory/InventorySystem.Equip.cs:165]: Ill add this and replace `IsInSameOrParentContainer` usages in this and other systems in a separate PR

**Pull request merged**